DalamudPlugin/RaidsRewritten/Scripts/Attacks/VoidGate.cs

```diff
@@ -0,0 +1,123 @@
+    public struct GateActor;
+    public struct AnimationActor;
+
+    private const float ResetBuffer = 5.0f;
+    private const string AbsorbVfx = "vfx/monster/c0101/eff/c0101wpinc0c.avfx";
+    private const string ExpelVfx = "vfx/monster/c0101/eff/c0101wpouc0c.avfx";
+    private const string GateActorVfx = "chara/monster/m0273/obj/body/b0001/vfx/eff/vm0001.avfx";
+    public static Entity CreateEntity(World world)
+    {
+        return FakeActor.Create(world)
+            .Set(new Position())
+            .Set(new Rotation())
+            .Set(new Scale())
+            .Set(new UniformScale(1f))
+            .Set(new Component())
+            .Set(new SpawnDelay())
+            .Set(new ExpelDelay())
+            .Add<Attack>();
+    }
+
+    public Entity Create(World world) => CreateEntity(world);
+
+    public void Register(World world)
+    {
+        world.System<Component, Position, Rotation, SpawnDelay, ExpelDelay>()
+            .Each((Iter it, int i, ref Component component, ref Position position, ref Rotation rotation, ref SpawnDelay spawnDelay, ref ExpelDelay expelDelay) =>
+            {
+                component.ElapsedTime += it.DeltaTime();
+
+                Entity animationActor;
+                Entity gateActor;
+
+                var entity = it.Entity(i);
+
+                using var animationQuery = world.QueryBuilder().With(flecs.EcsChildOf, entity).With<AnimationActor>().Build();
+                if (!animationQuery.IsTrue())
+                {
+                    animationActor = FakeActor.Create(it.World())
+                        .Set(new Position(position.Value))
+                        .Set(new Rotation(rotation.Value))
+                        .Add<AnimationActor>()
+                        .ChildOf(entity);
+                }
+                else
+                {
+                    animationActor = animationQuery.First();
+                }
+
+                using var gateQuery = world.QueryBuilder().With(flecs.EcsChildOf, entity).With<GateActor>().Build();
+                if (!gateQuery.IsTrue())
+                {
+                    gateActor = FakeActor.Create(it.World())
+                        .Set(new Position(position.Value))
+                        .Set(new Rotation(rotation.Value))
+                        .Add<GateActor>()
+                        .ChildOf(entity);
+                }
+                else
+                {
+                    gateActor = gateQuery.First();
+                }
+
+                switch (component.Phase)
+                {
+                    case Phase.Spawn:
+                        if (component.ElapsedTime < spawnDelay.DelayTime) { return; }
+                        AddActorVfx(animationActor, AbsorbVfx);
+                        DelayedAction.Create(world, () =>
+                        {
+                            AddActorVfx(gateActor, GateActorVfx);
+                        }, 0.5f);
```

Comment:
Delayed actions that operate on a system's entity need to be a child of the entity, or else if the entity is deleted before the delayed action executes, ECS will throw an exception crash
```suggestion
                        DelayedAction.Create(world, () =>
                        {
                            AddActorVfx(gateActor, GateActorVfx);
                        }, 0.5f).ChildOf(gateActor);
```